.github/workflows/process_data.py

```
@@ -56,11 +57,12 @@ def get_paginated_items(url, params):
     items = []
     while url:
         try:
-            response = requests.get(url, params=params)
+            # Add timeout to prevent hanging on slow/unresponsive servers
+            response = requests.get(url, params=params, timeout=30)
             response.raise_for_status()
         except requests.exceptions.RequestException as err:
-            logging.error(f"Error fetching items: {err}")
-            break
+            logging.error(f"Error fetching items from {url}: {err}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)